Apps/DiceRoller/DiceRoller/ViewController.m

```diff
@@ -30,22 +52,289 @@ - (void)didReceiveMemoryWarning{
+        
+        // loop through the project fields and store each data point in a dictionary
+        for (RProjectField *field in projectFields) {
+            
+            // see if this field contains "white" or "yellow" in it
+            if ([field.name.lowercaseString rangeOfString:@"white"].location != NSNotFound) {
+                
+                // if we're here, this is our value of the white dice
+                // remember, the object and key should be strings!
+                [dataRow setObject:[NSString stringWithFormat:@"%d", whiteDiceValue]
+                            forKey:[NSString stringWithFormat:@"%d", field.field_id.intValue]];
+                
+                
+            } else if ([field.name.lowercaseString rangeOfString:@"yellow"].location != NSNotFound) {
+                
+                // if we're here, this is our value of the yellow dice
+                [dataRow setObject:[NSString stringWithFormat:@"%d", yellowDiceValue]
+                            forKey:[NSString stringWithFormat:@"%d", field.field_id.intValue]];
+                
+            } else {
+                
+                // if we're here, this is the dice sum value
+                [dataRow setObject:[NSString stringWithFormat:@"%d", diceRollSum]
+                            forKey:[NSString stringWithFormat:@"%d", field.field_id.intValue]];
+                
+            }
+        }
+        
+        // add this data row to our data array
+        [rowMajorData addObject:dataRow];
+        
+        // at this point, feel free to add more data points to the rowMajorData array!
+        // to prevent the need for parsing the project's fields and logging in multiple times,
+        // make sure you only make these API calls once though.
+        
+        // when storing data, we typically follow the format we did above.  this style is called row-major
+        // because we stored one data row in our array at a time.  iSENSE, however, expects data in
+        // column-major format.  that is, our data object needs to be a dictionary where the keys are
+        // the field IDs and the values are an array of all data points for that key.  luckily, the iSENSE
+        // iOS library comes with a DataManager object that will perform this conversion step for us.
+        NSDictionary *columnMajorData = [DataManager convertDataToColumnMajor:rowMajorData forProjectID:[dm getProjectID]];
+        
+        // upload the data to iSENSE
+        [api uploadDataToProject:[dm getProjectID] withData:columnMajorData andName:dataSetName];
+        
+        // you're done! this uploadDataToProject method returns a data set ID number that you can store or check.
+        // this dataSetID variable will now contain either a positive integer if the data uploaded
+        // successfully (namely the ID of your new data set), or -1 if the upload failed.  if your upload failed,
+        // you may have misformatted the data
```

Comment:
At this point we may want some error checking, or even some basic user feedback.  For example, if the data set ID > 0, then the upload was likely successful, and we can show a brief waffle success message.  Otherwise display a waffle error message saying that the upload was unsuccessful.